entrypoint.sh

```diff
@@ -203,8 +203,56 @@ function start_postfix {
         echo "OpenDKIM: Add TXT record to DNS:"
         cat /var/db/dkim/default.txt
     fi
+
     
     sed -n -e '/^Domain\s/!p' -e '$aDomain '$HOSTNAME -i /etc/opendkim/opendkim.conf
+    # DKIM for all virtual domains and $HOSTNAME 
+    if [ "$SMF_DKIM_ALL" != "" ]; then
+        if [ ! -f /var/db/dkim/$HOSTNAME/default.private ]; then
+            echo "Moving ${HOSTNAME} keys to /var/db/dkim/$HOSTNAME/"
+            mkdir -p /var/db/dkim/$HOSTNAME
+            cp /var/db/dkim/default.* /var/db/dkim/$HOSTNAME
+        fi
+            chmod 400 /var/db/dkim/default.private
+            chown opendkim:opendkim /var/db/dkim/default.private
+            echo "Inserting ${HOSTNAME} data to /etc/opendkim/{KeyTable, SigningTable, TrustedHosts}"
+            echo "default._domainkey.${HOSTNAME} ${HOSTNAME}:default:/var/db/dkim/${HOSTNAME}/default.private
+            " >> /etc/opendkim/KeyTable
+
+            echo "${HOSTNAME} default._domainkey.${HOSTNAME}" >> /etc/opendkim/SigningTable
+
+            echo "${HOSTNAME}" >> /etc/opendkim/TrustedHosts
+        
+        for virtualDomain in $virtualDomains; do
```

Comment:
`HOSTNAME` is probably also in `$virtualDomains`, so I suspect that configuration for `HOSTNAME` is done twice.

Reply:
From what I understand from [here](https://github.com/dgraziotin/docker-simple-mail-forwarder/blob/master/entrypoint.sh#L68), `HOSTNAME` is not part of `$virtualDomains`. I did check for this empirically, to be sure. A folder `/var/db/dkim/${HOSTNAME}/default.private` was not created in a for loop over $virtualDomains. 

In any case, I think that this handles the case of reverting to a single domain (your PR), because HOSTNAME and all forwarded domains are kept separately.

Reply:
```
    - SMF_CONFIG=@example.com:xxx@gmail.com:pwd
    - SMF_DOMAIN=example.com
```
This is my config, based on https://github.com/dgraziotin/docker-simple-mail-forwarder/blob/99f27761e49e131a52b03159d69e7c652d25ffff/entrypoint.sh#L101 `virtualDomains` is a list of domains extracted from `emailFrom` which is the first element when split with `:`. I have to actually test but I'm quite sure that with my config, `example.com` is in `virtualDomains` and `SMF_DOMAIN`.

And what do you think, is #88 actually needed? Maybe dkim should always use a per-domain path style? Even if there is only one domain? This would mean removing [this](https://github.com/dgraziotin/docker-simple-mail-forwarder/blob/99f27761e49e131a52b03159d69e7c652d25ffff/entrypoint.sh#L194-L205) and removing  `SMF_DKIM_ALL` [here](https://github.com/dgraziotin/docker-simple-mail-forwarder/blob/99f27761e49e131a52b03159d69e7c652d25ffff/entrypoint.sh#L210).

Reply:
@petslane it was great that you pointed out your case, because it showcases my different use case! Luckily, both seem covered.

I think that we are both right: your `HOSTNAME` is part of `virtualDomains` because it is extracted from `SMF_CONFIG`. Normally people will have addresses `@${HOSTNAME}` in `SMF_CONFIG`. 

I do not. My `HOSTNAME` is dedicated to the SMTP server and does not appear in `SMF_CONFIG`. My domains use `HOSTNAME` in their MX entries. `HOSTNAME` is not used for sending or forwarding.

In light of this, I believe that both solutions should stay. #83 is needed for those with a single domain for everything (which I believe will be most people). #89 is for more complex cases. Everything is handled in `entrypoint.sh` and, if I did not make any mistake, things should stay clean all in all. @huan, what do you think?

Reply:
> In light of this, I believe that both solutions should stay. #83 is needed for those with a single domain for everything (which I believe will be most people). #89 is for more complex cases.

I have only one domain, why shouldn't I use your way of setting DKIM keys? From the user's point of view, it's currently complicated as there are 2 ways to set up DKIM, what option should the user choose? Why not have only one way to set up DKIM for SMF, your way, and there is no difference in how many domains are used (I expect that your way should work also with only 1 domain).

Reply:
I would be OK to have both solutions merged, but I am not sure I could deliver this in short. PR to the PR are welcome, I guess 😄